compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetUtils.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_AMD_AIE_TARGET_AIETARGETUTILS_H_
+#define IREE_AMD_AIE_TARGET_AIETARGETUTILS_H_
+
+#include <array>
+
+#include "mlir/IR/Types.h"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+struct AMDAIEUtils {
```

Comment:
I generally dislike such struct that are just carrying `static` methods. Its effectively just a `namespace`. Just drop the `struct` and have these as util functions. It doesnt need to like here though. This is information the compiler needs. The `Target` is more for managing the HAL. Just move this to `Transforms` folder or create a new `Utils` folder.

Reply:
Done. 

I like using structs like this because you get compile time errors instead of link time errors if the function in the .h and .cpp are slightly different. But users of APIs should come before developers of APIs, so I'm happy to make the change!